packages/mask/src/extension/popups/components/NetworkSelector/index.tsx

```diff
@@ -46,7 +46,7 @@ export const NetworkSelector = memo(() => {
     const networks = getRegisteredWeb3Networks()
     const account = useAccount()
     const chainId = useChainId()
-    const providerType = useProviderType()
+    const providerType = useValueRef(currentProviderSettings)
```

Comment:
Not `currentMaskWalletProviderSettings`?

Reply:
Yes, the logic here is to update the network synchronously when `currentMaskWalletProvider === currentProviderSettings`